### PR TITLE
added generated measure fields to zip file

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-5370](https://jira.cms.gov/browse/MAT-5370)
(Optional) Related Tickets:

### Summary
added generated measure fields to zip file

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Comment by jacobmainzer-sb** [comment]
There need to be some tests covering this functionality.

File: src/api/useMeasureServiceApi.tsx [hunk 1]
```diff
@@ -333,6 +333,26 @@ export class MeasureServiceApi {
     );
   }
 
+  async getZippedMeasureData(measureId: string): Promise<Blob> {
```
Comment (adongare) [inline src/api/useMeasureServiceApi.tsx]: I think the method name `getMeasureEport` or `exportMeasure` would match with the action performed.

File: src/components/measureList/MeasureList.tsx [hunk 2]
```diff
@@ -255,14 +253,23 @@ export default function MeasureList(props: {
     setOptionsOpen(false);
   };
 
-  const zipData = () => {
-    const zip = new JSzip();
-    zip.generateAsync({ type: "blob" }).then(function (content) {
-      saveAs(
-        content,
+  const zipData = async () => {
```
Comment (adongare) [inline src/components/measureList/MeasureList.tsx]: same this name as well

File: src/components/measureList/MeasureList.tsx [hunk 3]
```diff
@@ -255,14 +253,23 @@ export default function MeasureList(props: {
...
+      link.href = url;
+      link.setAttribute(
+        "download",
         `${targetMeasure.current?.ecqmTitle}-v${targetMeasure.current?.version}-${targetMeasure.current?.model}.zip`
       );
-    });
+      document.body.appendChild(link);
+      link.click();
+    } catch (err) {
+      return err;
```
Comment (adongare) [inline src/components/measureList/MeasureList.tsx:271]: should we do something with an error?

File: src/api/useMeasureServiceApi.tsx [hunk 4]
```diff
@@ -333,6 +333,26 @@ export class MeasureServiceApi {
...
+      );
+      return response.data;
+    } catch (err) {
+      const message = `Unable to zip the measure`;
+      console.error(message);
+      console.error(err);
+      throw new Error(message);
+    }
+  }
+
```
Comment (adongare) [inline src/api/useMeasureServiceApi.tsx:355]: nice to have a unit test

File: src/components/measureList/MeasureList.tsx (hunk 3 above)
Comment (adongare) [reply src/components/measureList/MeasureList.tsx:271]: nvm, seems out of scope.

Comment (sb-prateekkeerthi) [reply src/components/measureList/MeasureList.tsx:271]: Yeah, this was out of scope so I that is the reason I am just returning the error

File: src/api/useMeasureServiceApi.tsx (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/api/useMeasureServiceApi.tsx]: Thank you, I'll change it

File: src/components/measureList/MeasureList.tsx (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/components/measureList/MeasureList.tsx]: Thank you, I'll change it

File: src/api/useMeasureServiceApi.tsx (hunk 4 above)
Comment (sb-prateekkeerthi) [reply src/api/useMeasureServiceApi.tsx:355]: Sure, I'll add it

**Review by adongare — APPROVED** [review]

**Pull request merged**